build.sbt

```diff
@@ -401,6 +401,9 @@ def mimaSettings(moduleName: String) =
           exclude[MissingClassProblem](
             "cats.kernel.compat.scalaVersionMoreSpecific$suppressUnusedImportWarningForScalaVersionMoreSpecific"
           )
+        ) ++ //abstract package private classes
+        Seq(
+          exclude[DirectMissingMethodProblem]("cats.data.AbstractNonEmptyInstances.this")
```

Comment:
Are we sure this doesn't break BC? If we are sure maybe add a test in `binCompatTest`?  If not can we just add a separate method to provide the instance for `Align`

Reply:
I think that shall do. thanks! 
On a slightly related note, since we have been updating the build, I am a bit nervous that if something breaks `binCompatTest`, we wouldn't know. I'll create an issue.

Reply:
This just makes me vaguely uncomfortable. It's only used twice and it seems like it'd be almost equivalently  noisy just to write out the `new AbstractNonEmptyInstances[F, NEF] with Align[NEF]` in those two places.

I know we don't promise bincompat for people using Java, etc., but breaking it for the sake of saving a few lines doesn't feel ideal.